### ISSUE TITLE
Lava Fluid | Shortcodes cannot have non-String params (and it does not error)

### ISSUE THREAD

**Issue by jamrphy-bwd** [SparkDevNetwork/Rock#6009]
### Description

Consider a testparams shortcode with the following markup:
```liquid
<div>{{param1 | ToJSON }}</div>
<div>{{param2 | ToJSON }}</div>
<hr/>
```

On a page somewhere, the following:
```liquid
{% assign testVar = '' | AddToArray:'Hello, Rock!' | AddToArray:'This value should not be in testparams' %}

{[ testparams param1:'This works!' param2:'Yay!' ]}

{[ testparams param1:'Say null if this does not work:' param2:0 ]}

Correct second param value: {{ testVar[0] | ToJSON }}
{[ testparams param1:'Should say Hello, Rock:' param2:testVar[0] ]}
``` 

### Actual Behavior

The above lava outputs:
![image](https://github.com/user-attachments/assets/20d89f14-b7bc-4c73-acdb-35fb2c299bf1)

### Expected Behavior

Values should be passed. Alternatively, the lava template should error.

I noticed in #3605  ([this comment](https://github.com/SparkDevNetwork/Rock/issues/3605#issuecomment-813616559)) that the topic of params has come up before with the notable conclusion that shortcode parameters are meant for simple inputs, double quotes are not valid syntax, and single quotes are required (else pass variables).

If single quotes or variable parameters are the requirements/limitations, that's fine. It's not a difficult workaround.

However, in that case (and in any case where there are extraneous, unprocessed symbols within language constructs), shouldn't the template give an error?

This issue is also somewhat related to #6008, given the behavior with the attempted array access instead matching on the array var.

### Steps to Reproduce

- Make shortcode
- Try to pass non-string, non-variable
- See nothing passed and no error

### Issue Confirmation

- [X] Perform a search on the [Github Issues](https://github.com/issues?q=is%3Aissue+user%3ASparkDevNetwork+-repo%3ARock) to see if your bug or enhancement is already reported.
- [X] Reproduced the problem on a fresh install or on the [demo site](https://rock.rocksolidchurchdemo.com/).

### Rock Version

16.5

### Client Culture Setting

en-US

**Comment by chead4** [comment]
@jamrphy-bwd To help us investigate this issue, it would be greatly appreciated if you could clarify the input and output of the shortcode test that is the issue.

**Comment by jamrphy-bwd** [comment]
Sure! Let me try to clarify:

Related but distinct issues on these two lines (really just paying attention to `param2`):
```liquid
{[ testparams param1:'Say null if this does not work:' param2:0 ]}
{[ testparams param1:'Should say Hello, Rock:' param2:testVar[0] ]}
```

The first line fails to pass the value of `0` to the shortcode context for `param2` (no `param2` variable will exist within the shortcode). Per discussion on #3605, shortcodes "require that parameters be enclosed with single quotes." Thus, the first line is invalid syntax. It looks reasonable, but it is invalid syntax that fails silently. In other contexts, Fluid errors on malformed syntax, so this behavior seems "bug-ish".

Regarding the second line, a regex matches `testVar` (#6008 displays the code that does so), and it passes the entire array rather than the value of the specific index `testVar[0]`. Like the first line, this one fails silently and is technically invalid syntax, but it does so while also passing data the developer did not intend to pass to the shortcode.

In summary, I'd say the main _issue_ is that Fluid is not erroring on the invalid syntax like it usually does. Being strict on syntax is a very good thing as it prevents developers from trying to troubleshoot problems at the wrong level of abstraction.

IMO, the limitations on the syntax feel a little unusual (it makes shortcodes feel like an incomplete language extension), but in the context of a GH issue, I'm trying to articulate the aspects that feel the most bug-ish rather than feature-request-ish. :)
